main_window/rocket_data.py

```diff
@@ -157,19 +157,22 @@ def last_value_and_time(self, device: DeviceType, data_entry_id: DataEntryIds) -
         :return: Value, Time
         :rtype:
         """
-        ret = self.time_series_by_device(device, data_entry_id)
-
-        if ret is None:
-            return None
+        with self.data_lock:
+            all_times = list(self.timeset.keys())
 
-        t, y = ret
+            full_address = self.device_manager.get_full_address(device)
+            if full_address is None:
+                return None
```

Comment:
Not going to require it, but my understanding is that `None` is acting like an error code, but a more Pythonic approach might be to raise a (custom) DataUnavailable exception.